### PR TITLE
loop search optimization (#91)

### PR DESCRIPTION
**Pull request by emreesendir**
New feature: No matches found message added.
Bugfix: setSearchTerm() at line 147 was causing problems. Instead, now it finds matches and triggers re-render when it is done.
Bugfix: When the user clears the input, it was still showing the last result. Now it clears the result as well.

### CODE REVIEW

**Review by giuliamml — COMMENTED** [review]
Looks good to me.

File: frontend/src/pages/FindChain.tsx [hunk 1]
```diff
@@ -144,14 +143,31 @@ const FindChain = () => {
...
+    //find matches
+    let matches: IChain[] = [];
+    let matchesLength: number = 0;
     chainData.filter((val) => {
-      if (searchTerm == "") {
-      } else if (val.name.toLowerCase().includes(searchTerm.toLowerCase())) {
-        return setValue(val);
+      if (val.name.toLowerCase().includes(searchTerm.toLowerCase())) {
+        matches.push(val);
+        matchesLength++;
```
Comment (timstokman) [inline frontend/src/pages/FindChain.tsx:159]: Please use .filter correctly: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter

matches = chainData.filter(val => val.name.toLowerCase().includes(searchTerm.toLowerCase()));

Comment (emreesendir) [reply frontend/src/pages/FindChain.tsx]: Thank you for the correction. I go with the find method here then.

**Comment by emreesendir** [comment]
> This approach is ok when there's not a lot of data/chains, but you'll want to filter server-side if the amount of data gets bigger.

I don't know how to proceed here. I just tried to stay as close as to the original. How do you think I should proceed? Should I do it now or leave it for another time? @timstokman @giuliamml

**Review by timstokman — APPROVED** [review]

**Pull request merged**